datafusion/datasource-parquet/src/row_filter.rs

```diff
@@ -662,14 +660,11 @@ mod test {
         assert!(!can_expr_be_pushed_down_with_schemas(
             &expr,
             &file_schema,
-            &table_schema
         ));
     }
 
     #[test]
-    fn projected_columns_prevent_pushdown() {
-        let table_schema = get_basic_table_schema();
-
+    fn projected_or_partition_columns_prevent_pushdown() {
```

Comment:
This test actually tests for this... except that since both schemas were being passed in it hit the bug as well.